File: Day-01/python/RainbowDashLabs/day_1.py
with open('puzzle.txt') as f:
    depths = [int(i) for i in f.readlines()]

last = None


def eval(curr_last, curr):
    global last
    last = curr
    return 0 if curr_last is None else (1 if curr_last < curr else 0)


increased = sum([eval(last, depth) for depth in depths])

print(f"Part one {increased}")

frame = []
last = 0
increased = 0

for depth in depths:
    last = sum(frame)
    frame.append(depth)
    if len(frame) > 3:
        frame.pop(0)
        increased += 1 if last < sum(frame) else 0

print(f"Part two {increased}")

File: Day-01/java/RainbowDashLabs/Day1.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayDeque;
import java.util.stream.Collectors;

public class Day1 {
    public static void main(String[] args) throws IOException {
        var inputs = Files.readAllLines(Path.of("puzzle.txt")).stream().map(Integer::parseInt).collect(Collectors.toList());

        var last = Integer.MAX_VALUE;
        var increased = 0;
        for (var input : inputs) {
            if (last < input) increased++;
            last = input;
        }

        System.out.println("First part " + increased);

        increased = 0;
        var frame = new ArrayDeque<Integer>();
        last = 0;
        for (var input : inputs) {
            last = frame.stream().mapToInt(value -> value).sum();
            frame.add(input);
            if(frame.size() > 3){
                frame.pop();
                if(last < frame.stream().mapToInt(value -> value).sum()) increased++;
            }
        }
        System.out.println("Second part " + increased);
    }
}
